cpp/src/arrow/util/async_util.h

```diff
@@ -277,6 +278,8 @@ class ARROW_EXPORT ThrottledAsyncTaskScheduler : public AsyncTaskScheduler {
   /// Allows task to be submitted again.  If there is a max_concurrent_cost limit then
   /// it will still apply.
   virtual void Resume() = 0;
+  /// Return the number of tasks queued but not yet submitted
+  virtual std::size_t QueueSize() = 0;
```

Comment:
Would this better a `std::size_t QueueSize() const`?

Reply:
The implementation uses a `std::mutex` and so I'd have to mark the mutex mutable right?  Which would you prefer?  "mutable mutex" or "non-const accessor"?  I don't have strong preference.

Reply:
Usally mutex is used with mutable. But this LGTM either, I don't have strong preference too